### DISCUSSION TITLE
怎么断言一个值等于 0 或者 1

### DISCUSSION THREAD

**Discussion by a2asister** [openbuildxyz/mina-zkapp-bootcamp#236] [Q&A] [answered]
怎么断言一个值等于 0 或者 1

**Comment by logere** [comment]
很优雅的实现判断是否是0 https://github.com/iden3/circomlib/blob/master/circuits%2Fcomparators.circom#L24

**Reply by a2asister** [reply]
这个怎么引用呢？

**Reply by logere** [reply]
判断一个 Field 是否为 0, 并返回判断值 out
```js
    const in_ = Field(10086);
    const in_val = in_.toBigInt();
    const in_inv = in_val === BigInt(0)? Field(0) : in_.inv();
    console.log('in_inv:', in_inv.toBigInt());
    const out = Field(1).sub(in_inv.mul(in_));
    in_.mul(out).assertEquals(Field(0));
```

有了这个之后, 判断一个 $\text{in} \in \{ a_0, a_1, ..., a_n \}$ 等价于判断 $(in - a_0)(in -a_1) \cdots (in-a_n)  = 0$.

**Reply by logere** [reply]
好像可以直接用用 equals 函数

**Comment by linghuccc** [comment]
如果只有 0 或者 1 这 2 种可能性，建议直接用 Bool 类型

**Comment by lispking** [comment, marked as answer]
Bool.or(vaule.equals(Field(0)), vaule.equals(Field(1))).assertTrue('Invaild value');
